### PR TITLE
Bash issue fix

### LINKED ISSUE #40: If buildfile errors during Plan creation, there is an uncaught exception when trying to create the summary table
If a plan is created and has no default tasks or the user specifies a task that does not exist, the build will error appropriately, but there is an uncaught exception that follows. See screenshots below:

Screenshot 2024-07-15 151305

Screenshot 2024-07-15 151336

### PR DESCRIPTION
**Pull request by nbhoski**
This PR closes #40 and #42 
* Added file check before creating build summary so that it does not hit the exception if plan failure occures

### CODE REVIEW

File: .github/workflows/bat.yml [hunk 1]
```diff
@@ -183,4 +183,19 @@ jobs:
         run: |
           grep "::group::deploy" console.log
           grep "::error::error task failed" console.log
+          rm console.log
```
Comment (mw-hrastega) [inline .github/workflows/bat.yml:186]: Are the text elements in this file user-visible?

File: src/buildSummary.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2024 The MathWorks, Inc.
```
Comment (mw-hrastega) [inline src/buildSummary.ts:1]: The messages in this file look good.

File: src/buildSummary.ts [hunk 3]
```diff
@@ -43,20 +43,26 @@ export function processAndDisplayBuildSummary() {
...
     const filePath: string = join(runnerTemp, `buildSummary${runId}.json`);
     let taskSummaryTableRows;
-    try {
-        const data = JSON.parse(readFileSync(filePath, { encoding: 'utf8' }));
-        taskSummaryTableRows = getBuildSummaryTable(data);
-    } catch (e) {
-        console.error('An error occurred while reading the build summary file:', e);
-        return;
-    } finally {
+    if (checkFileExists(filePath)) {
```
Comment (davidbuzinski) [inline src/buildSummary.ts:46]: Can use import and use `fs.existsSync` and remove the custom `checkFileExists` definition. Can also remove `accessSync` from import list.
```suggestion
    if (existsSync(filePath)) {
```

Comment (nbhoski) [reply src/buildSummary.ts:46]: I saw exist deprectaed in chat gpt hence did not use it :) shall update it

File: .github/workflows/bat.yml (hunk 1 above)
Comment (nbhoski) [reply .github/workflows/bat.yml:186]: no these are not visible

File: src/buildSummary.ts [hunk 4]
```diff
@@ -43,20 +43,26 @@ export function processAndDisplayBuildSummary() {
...
+        } finally {
+            try {
+                unlinkSync(filePath);
+            } catch (e) {
+                console.error(`An error occurred while trying to delete the build summary file ${filePath}:`, e);
+            }
         }
+        writeSummary(taskSummaryTableRows);
+    } else {
+        core.info(`Build summary data not created.`);
```
Comment (nbhoski) [inline src/buildSummary.ts:63]: @mw-hrastega let me know if these exceptions looks good

File: src/buildSummary.ts [hunk 5]
```diff
@@ -43,20 +43,26 @@ export function processAndDisplayBuildSummary() {
...
     let taskSummaryTableRows;
-    try {
-        const data = JSON.parse(readFileSync(filePath, { encoding: 'utf8' }));
-        taskSummaryTableRows = getBuildSummaryTable(data);
-    } catch (e) {
-        console.error('An error occurred while reading the build summary file:', e);
-        return;
-    } finally {
+    if (existsSync(filePath)) {
+
```
Comment (sameagen-MW) [inline src/buildSummary.ts:47]:
```suggestion
```

Extra line here.

**Review by sameagen-MW — APPROVED** [review]

**Review by mw-hrastega — APPROVED** [review]

**Comment by nbhoski** [comment]
This PR also closes #44

**Pull request merged**